src/test/java/org/dsmhack/controller/LoginControllerTest.java

```diff
@@ -36,22 +37,37 @@
 
     @Before
     public void setup() {
-        when(userRepository.findByEmail(anyString())).thenReturn(new User());
         mockMvc = MockMvcBuilders.standaloneSetup(loginController).build();
     }
 
     @Test
     public void loginReturns201() throws Exception {
+        when(userRepository.findByEmail(anyString())).thenReturn(new User());
+
         mockMvc.perform(post("/login/sendCode")
                 .contentType(MediaType.APPLICATION_JSON)
-                .content("email address"))
+                .content("test@aol.com"))
                 .andExpect(status().isCreated());
     }
 
     @Test
     public void loginCallsLoginServiceWithEmailAddress() throws Exception {
+        when(userRepository.findByEmail(anyString())).thenReturn(new User());
+
         loginController.login("test@aol.com");
         verify(loginService).login(any(User.class));
     }
 
+    @Test
+    public void emailDoesNotExist() throws Exception {
+        when(userRepository.findByEmail(anyString())).thenReturn(null);
+
+        MvcResult mvcResult = mockMvc.perform(
+                post("/login/sendCode")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content("test@aol.com")
+        ).andExpect(
+                status().isNotFound()
+        ).andReturn();
+    }
 }
```

Comment:
omg missing newline!
https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

Reply:
Lol, I'll add one.